NitroxClient/GameLogic/Spawning/InstalledBatteryEntitySpawner.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using NitroxClient.Communication;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures.GameLogic.Entities;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Packets;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning;
+
+public class InstalledBatteryEntitySpawner : EntitySpawner<InstalledBatteryEntity>
+{
+    public override IEnumerator SpawnAsync(InstalledBatteryEntity entity, TaskResult<Optional<GameObject>> result)
+    {
+        Optional<GameObject> parent = NitroxEntity.GetObjectFrom(entity.ParentId);
+
+        if (!parent.HasValue)
+        {
+            Log.Info($"Unable to find parent to install battery {entity}");
+            result.Set(Optional.Empty);
```

Comment:
If we are setting the results here do we not want to return? If we continue we will send up always having the result set further down the method

Reply:
good catch, forgot to return in case the object is not in a good state.